batteryMangementSystem_Checker.c

```diff
@@ -0,0 +1,78 @@
+#include <stdio.h>
+#include <assert.h>
+#include "BmsData.h"
+
+char* BmsPrameters[] = {"Temperature", "State of Charge", "Charging Rate"};
+
+int batteryIsOk(float temperature, float soc, float chargeRate) {
+	int temperatureStatus=0;
+	int SocStatus=0;
+	int ChargeRateStatus=0;
+	temperatureStatus=IsTemperatureOk(temperature);
+	SocStatus=IsSocOk(soc);
+	ChargeRateStatus=IsChargeRateOk(chargeRate);
+	int IsBattaeryOkValue=(temperatureStatus&SocStatus&ChargeRateStatus);
```

Comment:
Usage of bitwise and (single &). Is that the expected code or should it be && (logical &)?

Reply:
yes it can be &&

Reply:
usage of bitwise & is specifically used for bitwise operations. You need to strongly guard your code to return only 0 or 1. Any other non zero value could lead to errors in future and it will be difficult to trace.

here the requirement is for logical evaluation of expressions, hence from maintainability and readability point of view && is best suited.

Reply:
@prithamvshetty i will correct that thank You !!